### DISCUSSION TITLE
Does Replcache handles split brain

### DISCUSSION THREAD

**Discussion by Huthesh** [belaban/JGroups#669] [General]
HI Team,

We are using ReplCache for a simple cluster of 3 nodes. Nodes are hosted on Kubernetes and we are using Jgroups Kubernetes for forming the cluster.

My question is how does ReplCache handle Split brain?

**Comment by belaban** [comment]
On 10.10.22 05:05, Huthesh wrote:
> HI Team,
> 
> We are using ReplCache for a simple cluster of 3 nodes. Nodes are hosted 
> on Kubernetes and we are using Jgroups Kubernetes for forming the cluster.
> 
> My question is how does ReplCache handle Split brain?

It doesn't handle partitions: nodes on either side of a partition can be 
updated separately, leading to data inconsistencies.

When a network partition heals, data should be consistent again, but at 
the cost of one side overwriting data of another side.

If you want consistency and partition handling, take a look at jgroups-raft.
Cheers


> Reply to this email directly, view it on GitHub 
> <https://github.com/belaban/JGroups/discussions/669>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AADPZXNQIVBDGEOHDO77PVLWCOBYRANCNFSM6AAAAAARA62Q2Q>.
> You are receiving this because you are subscribed to this thread.Message 
> ID: ***@***.***>
> 

-- 
Bela Ban | http://www.jgroups.org

**Reply by Huthesh** [reply]
@belaban Does ReplCache work over jgroups-raft?

**Comment by belaban** [comment]
Bela BanNo. Use the equivalent classes, e.g. ReplicatedStateMachine-------- Original message --------From: Huthesh ***@***.***>Date: Wed, 12 Oct 2022, 14:52To: belaban/JGroups ***@***.***>Cc: Bela Ban ***@***.***>, Mention ***@***.***>Subject: Re: [belaban/JGroups] Does Replcache handles split brain (Discussion #669)
@belaban Does ReplCache work over jgroups-raft?
